### DISCUSSION TITLE
DOS 1 question

### DISCUSSION THREAD

**Discussion by telanus** [640-KB/GLaBIOS#23] [News]
I added the bios to my local build of 86Box, and works great. 

Started trying it to see what run and what doesn't. Started with MS-DOS and IBM-DOS, both release 1. For IBM-DOS I couldn't get one of the disk images that I have to work, while with MS-DOS I could only get one to work. Images work with the generic xt bios in 86Box.

So does the  bios support dos series 1

**Comment by 640-KB** [comment]
Glad to hear it's working great for you!  Yes, it does support DOS 1, with one small note.  When booting from floppy disk GLaBIOS will (by default) attempt to verify a valid boot signature on the [MBR (Master Boot Record)](https://en.wikipedia.org/wiki/Master_boot_record#Disk_partitioning) before attempting to boot from the disk.  However, the standard MBR and boot signature was not introduced until DOS 2.0, so for DOS 1 it would consider that invalid. There is a [build time option](https://github.com/640-KB/GLaBIOS/blob/0d215eb173f2802de3dd4ec66d2d9934e068e4dd/src/GLABIOS.ASM#L242) to enable/disable that checking, which will make DOS 1 boot.

It's good point though that while it seems like a good idea to validate the MBR, perhaps having it not work by default on DOS 1 isn't the expected behavior and should be disabled for at least 5150 and 5160 builds going forward. At the very least, I'll make it a more prominent FAQ or something.

Here are updated builds for [5150/PC](https://github.com/640-KB/GLaBIOS/releases/download/v0.2.1/GLABIOS_0.2.1_8P_NOMBR.ROM) and [5160/XT](https://github.com/640-KB/GLaBIOS/releases/download/v0.2.1/GLABIOS_0.2.1_8X_NOMBR.ROM) with the MBR check disabled.

Let me know if that does the trick!

640KB

**Reply by telanus** [reply]
Thanx these work with Early pre-dos2 images (also CP/M 1.0 & 1.1)

**Reply by 640-KB** [reply]
That's great to hear!  I am going to remove the check for the PC/XT builds since it is the more compatible way.  

Thanks so much for finding that and bringing it up!

640KB

**Reply by 640-KB** [reply]
@telanus This has been added in the 0.2.2 release for 5150.  This version has a number of small updates as well, so this is the one you want! Thanks again for checking it out and the suggestion!

https://github.com/640-KB/GLaBIOS/releases/tag/v0.2.2

**Reply by telanus** [reply]
Quick Question. The nombr change is only for the 5150/PC and not the XT/5160?

**Reply by 640-KB** [reply]
> Quick Question. The nombr change is only for the 5150/PC and not the XT/5160?

Well, originally yes.  As I dug into it my thinking was that DOS 1.x was only ever intended to be used on 5150 and only DOS 2 and above would have been used on 5160's since hard drives were standard at that time.

Then I started looking further, it turns out that there were [5160's sold without hard drives](http://www.bitsavers.org/pdf/ibm/pc/SA38-0037-00_Personal_Computer_Family_Service_Information_Manual_Jul89.pdf) later on (such as the model 267, 268, 277, etc).

It then finally hit me that the entire check is pointless because when any hard drive controller is installed, it will re-vector `INT 19h` meaning that the GLaBIOS MBR check wouldn't happen anyway, so most likely isn't serving the original intent anyway and only making it more difficult to run DOS 1.0 and potentially other disk images that do not contain MBR.

Long story short...  I've changed MBR check to now be disabled by default and updated the 5160 ROM images on the releases page so now it is both, as well as all subsequent ones have changed.
